### ISSUE TITLE
How to speed up the cleaning process when using CleanLab with other ensemble classifier?

### ISSUE THREAD

**Issue by cannolis** [cleanlab/cleanlab#539]
Hello, 
When I used CleanLab with other ensemble classifiers in sklearn under high class imbalance, I got the results after a long time, so I want to find out if there are some methods to speed up the cleaning process.

**Comment by elisno** [comment]
Hi, @cannolis.

Could you provide more details?

1. Can you say what kind of results you're looking for?
2. What "other" ensemble classifiers are you referring to?
3. What cleanlab functionality are you using?

It would be easier to give you answers if you provide a concrete Minimum Working Example that demonstrates this behavior, so we can see if there are 

Without any context, I guess you could fiddle around with the hyperparameters in the classifiers that you chose if you intend to re-run everything from scratch.

**Comment by cgnorthcutt** [comment]
@cannolis the trick is to do it in two steps.

Instead of using `CleanLearning`ehich requires training the model everytime you instantiate, step 1: pre-compute your predicted probabilities on your dataset using cross validation (only once), then step 2: run `cleanlab.filter.find_label_issues` as many times as you like with various hyper parameters (it should take only seconds, and uses the predicted probabilities as input so no need to re-run your model).

That's the fast workflow.

Consider playing with the `frac_noise` and `filter_method` hyper-parameters. 

Have fun!! Let us know if you run into any issues. If you need to correct the labels or you're looking for a no-code, automated solution, use [Cleanlab Studio](https://cleanlab.ai/studio).
